tests/gabbits/create-snippet-errors.yaml

```diff
@@ -0,0 +1,59 @@
+fixtures:
+  - XSnippetApi
+
+tests:
+  - name: create a new snippet (invalid method)
+    PUT: /v1/snippets
+    request_headers:
+      content-type: application/json
+    data:
+      title: Hello, World!
+      syntax: python
+      content: print('Hello, World!')
+      tags:
+        - spam
+        - eggs
+    status: 404
+
+  - name: create a new snippet (invalid JSON)
+    POST: /v1/snippets
+    request_headers:
+      content-type: application/json
+    data: spam
+    status: 400
+
+  - name: create a new snippet (empty content)
+    POST: /v1/snippets
+    request_headers:
+      content-type: application/json
+    data:
+      title: Hello, World!
+      syntax: python
+      content: ""
+      tags:
+        - spam
+        - eggs
+    status: 400
+    response_headers:
+      content-type: application/json
+    response_json_paths:
+      $:
+        message: "`content` - empty values not allowed."
+
```

Comment:
I have two ideas for tests:

1. Test when `tags` receive a string, not list of strings.
2. Test when unexpected property is passed, e.g. `syntax1`.